str.go

```diff
@@ -0,0 +1,74 @@
+package merkletree
+
+import (
+	"github.com/coniks-sys/libmerkleprefixtree-go/crypto"
+	"github.com/coniks-sys/libmerkleprefixtree-go/internal"
+)
+
+// SignedTreeRoot represents a signed tree root, which is generated
+// at the beginning of every epoch.
+// Signed tree roots contain the current root node,
+// the current and previous counters, the hash of the
+// previous STR, and its signature.
+// STR should be final
+type SignedTreeRoot struct {
+	tree        *MerkleTree
+	counter     int64
+	prevCounter int64
+	prevStrHash []byte
+	sig         []byte
+	policies    Policies
+}
+
+func NewSTR(key crypto.SigningKey, policies Policies, m *MerkleTree, counter int64, prevHash []byte) *SignedTreeRoot {
+	str := &SignedTreeRoot{
+		tree:        m,
+		counter:     counter,
+		prevCounter: counter - 1,
+		prevStrHash: prevHash,
+		policies:    policies,
+	}
+	bytesPreSig := str.getSTRBytesForSig()
+	str.sig = crypto.Sign(key, bytesPreSig)
+	return str
+}
+
+func (str *SignedTreeRoot) getSTRBytesForSig() []byte {
+	var strBytes []byte
+	strBytes = append(strBytes, util.LongToBytes(str.counter)...)     // t - counter number
+	strBytes = append(strBytes, util.LongToBytes(str.prevCounter)...) // t_prev - previous counter number
+	strBytes = append(strBytes, str.tree.root.serialize()...)         // root
```

Comment:
I don't think so, according to the reference implementation: 
https://github.com/coniks-sys/coniks-ref-implementation/blob/master/coniks_server/src/org/coniks/coniks_server/ServerUtils.java#L364

The reason for including the `treeHash` field into the `MerkleTree` is discussed here:
https://github.com/coniks-sys/libmerkleprefixtree-go/pull/8#discussion_r68999957

But now I'm thinking that I should remove the `treeHash` field and move the `GetHash` method to the test source file to avoid confusing.